octopus/src/test/scala/octopus/AsyncValidationRulesSpec.scala

```diff
@@ -36,7 +43,7 @@ class AsyncValidationRulesSpec
   private val Email_Err_LeftCase = "Invalid email left case"
   private val Exception_HandledDuringValidation = "Exception handled during validation"
 
-  private val userValidator = Validator[User].async
+  private val userValidator = Validator[User].async[Future]
```

Comment:
Seems to me like a breaking change.
What do you think to keep compatibility and when calling `.async` we assume we still use Futures, but for custom wrappers we introduce `.asyncM` which requires explicit wrapper passing, like `.asyncM[Task]`, `.asyncM[IO]`?

Reply:
Done